### PR TITLE
GUACAMOLE-234: Fix resource leaks in new LDAP code.

### PR DESCRIPTION
**Pull request by necouchman**
I believe I've got the two resource leak issues resolved from the new LDAP code...

### CODE REVIEW

File: extensions/guacamole-auth-ldap/src/main/java/org/apache/guacamole/auth/ldap/ObjectQueryService.java [hunk 1]
```diff
@@ -223,10 +224,11 @@ else if (results.isReferral() && request.isFollowReferrals()) {
                 
             }
 
+            results.close();
```
Comment (mike-jumper) [inline extensions/guacamole-auth-ldap/src/main/java/org/apache/guacamole/auth/ldap/ObjectQueryService.java]: Will this still leak if an exception is thrown above?

Comment (mike-jumper) [reply extensions/guacamole-auth-ldap/src/main/java/org/apache/guacamole/auth/ldap/ObjectQueryService.java]: [`try`-with-resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) might be an option.

Comment (necouchman) [reply extensions/guacamole-auth-ldap/src/main/java/org/apache/guacamole/auth/ldap/ObjectQueryService.java]: Okay, gave this another shot with a `try...finally` block.  I don't like the nested `try` statements - I could declare the `results` variable outside of the first `try` block, instead, if you think that would be a better route?

Comment (mike-jumper) [reply extensions/guacamole-auth-ldap/src/main/java/org/apache/guacamole/auth/ldap/ObjectQueryService.java]: Well, I think you can likely avoid all that with Java 8 fanciness (`try`-with-resources). Something like:

```java
// Search within subtree of given base DN
LdapConnectionConfig ldapConnectionConfig = ldapConnection.getConfig();
SearchRequest request = ldapService.getSearchRequest(baseDN, query);

try (SearchCursor results = ldapConnection.search(request)) {
    ...
}
catch (CursorException | IOException | LdapException e) {
    ...
}
```

Comment (mike-jumper) [reply extensions/guacamole-auth-ldap/src/main/java/org/apache/guacamole/auth/ldap/ObjectQueryService.java]: In fact ... I think the same implicit `close()` from `try`-with-resources would work with `LdapNetworkConnection`, as well. It implements `AutoCloseable` just as `SearchCursor` does.

Comment (necouchman) [reply extensions/guacamole-auth-ldap/src/main/java/org/apache/guacamole/auth/ldap/ObjectQueryService.java]: Cool - I'll rework it, give me 5 minutes :smile:

**Comment by necouchman** [comment]
@mike-jumper: Okay, reworked both those instances with `try...finally` blocks - let me know if that looks okay.  It'll change the error reporting in the method from `LDAPConnectionService` that was changed, but I think it's still acceptable.

**Pull request merged**